php/libraries/Candidate.class.inc

```diff
@@ -273,15 +274,20 @@ class Candidate implements \LORIS\StudyEntities\AccessibleResource,
                 EDC_NOT_SPECIFIED
             );
         }
-        if ($dateOfBirth !== null
-            && (DateTime::createFromFormat('Y-m-d', $dateOfBirth) === false
-            || empty($dateOfBirth))
-        ) {
+        if (empty($dateOfBirth)) {
             throw new InvalidArgumentException(
                 "Date of Birth must be specified",
                 DOB_NOT_SPECIFIED
             );
         }
+        $dob = DateTime::createFromFormat('Y-m-d', $dateOfBirth);
+        if (!empty($dob->getLastErrors())) {
+            error_log(print_r($dob->getLastErrors(), true));
```

Comment:
I think that getLastError is static. It should be called using `\DateTime::getLastErrors()`

Reply:
Looking at documentation
https://www.php.net/manual/en/datetime.getlasterrors.php
It returns an array or false so `!empty` is always true.

Reply:
I tried it and when there are no errors, it returns the following:
```
array(4) {
  ["warning_count"]=>
  int(0)
  ["warnings"]=>
  array(0) {
  }
  ["error_count"]=>
  int(0)
  ["errors"]=>
  array(0) {
  }
}
```
That is not empty.
 You were right in checking for `error_count === 0 and warning_count === 0`